src/Shared/Recruit.Vacancies.Client/Infrastructure/Reports/ReportService.cs

```diff
@@ -58,15 +58,16 @@ public async Task GenerateReportAsync(Guid reportId)
                 _logger.LogInformation("Report: {reportId} in progress.", reportId);
 
                 var reportStrategy = _reportStrategyAccessor(report.ReportType);
-                var data = await reportStrategy.GetReportDataAsync(report.Parameters);
+                var reportStrategyResult = await reportStrategy.GetReportDataAsync(report.Parameters);
 
                 report = await _reportRepository.GetReportAsync(reportId);
                 if (report.Status == ReportStatus.Generated)
                 {
                     _logger.LogInformation("Report: {reportId} already generated in the meantime. Ignoring.", reportId);
                     return;
                 }
-                report.Data = data;
+                report.Headers = reportStrategyResult.Headers;
```

Comment:
What about existing Provider reports won't Headers be null? I'm not sure about this `Report.Header` field.  Can't we just pass in a per-report strategy/delegate for headers when we generate the report data?